fv3core/stencils/d_sw.py

```diff
@@ -153,29 +146,67 @@ def heat_damping_term(ub, vb, gx, gy, rsin2, cosa_s, u2, v2, du2, dv2):
 
 
 @gtstencil()
-def heatdamping(
-    ub: sd,
-    vb: sd,
-    delp: sd,
-    fx: sd,
-    fy: sd,
-    gx: sd,
-    gy: sd,
-    rsin2: sd,
-    cosa_s: sd,
-    heat_source: sd,
-    diss_est: sd,
-    damp: float,
-    do_skeb: int,
+def heat_damping(
+    ub: FloatField,
+    vb: FloatField,
+    ut: FloatField,
+    vt: FloatField,
+    u: FloatField,
+    v: FloatField,
+    delp: FloatField,
+    rsin2: FloatField,
+    cosa_s: FloatField,
+    rdx: FloatField,
+    rdy: FloatField,
+    heat_source: FloatField,
+    dissipation_estimate: FloatField,
+    kinetic_energy_fraction_to_damp: float,
+    calculate_dissipation_estimate: int,
 ):
+    """
+    Calculates heat source from vorticity damping implied by energy conservation.
+
+    Args:
+        ub (in)
+        vb (in)
+        ut (in)
+        vt (in)
+        u (in)
+        v (in)
+        delp (in)
+        rsin2 (in)
+        cosa_s (in)
+        rdx (in): radius of Earth multiplied by x-direction gridcell width
+        rdy (in): radius of Earth multiplied by y-direction gridcell width
+        heat_source (out): heat source from vorticity damping
+            implied by energy conservation
+        diss_est (out): dissipation estimate, only calculated if
+            calculate_dissipation_estimate is 1
+        kinetic_energy_fraction_to_damp (in): according to its comment in fv_arrays,
+            the fraction of kinetic energy to explicitly damp and convert into heat.
+            TODO: confirm this description is accurate, why is it multiplied
+            by 0.25 below?
+        calculate_dissipation_estimate (in): If 1, calculate dissipation estimate.
+            Equivalent in Fortran model is do_skeb
+    """
     with computation(PARALLEL), interval(...):
+        ubt = (ub + vt) * rdx
+        fy = u * rdx
```

Comment:
@jdahm from an optimization perspective, is there any penalty to removing fx/fy entirely and replacing it with e.g. u * rdx? the latter is much more clear, I'm sure these temporaries were only made to manually optimize the code.

Reply:
If it’s not already read in the computation elsewhere, it will generate an extra read, but that is really minor compared to the other issues we are facing. Go for clarity!